nixos/modules/services/mail/dovecot.nix

```diff
@@ -288,6 +395,26 @@ in
   options.services.dovecot2 = {
     enable = mkEnableOption "the dovecot 2.x POP3/IMAP server";
 
+    package = mkPackageOption pkgs "dovecot" {
+      default = [
+        (if versionAtLeast config.system.stateVersion "25.11" then pkgs.dovecot else pkgs.dovecot_2_3)
+      ];
+    };
+
+    configVersion = mkOption {
+      default =
+        if versionAtLeast (majorMinor cfg.package.version) "2.4" then cfg.package.version else null;
```

Comment:
This probably needs a `defaultText`. Otherwise the docs has to eval `cfg.package.version`.